<?xml version="1.0"?>
<robot name="delta_robot_ros2" xmlns:xacro="http://ros.org/wiki/xacro">

  <!-- Include common macros and motor definition -->
  <xacro:include filename="$(find delta_robot_ros2)/urdf/common_macro.xacro"/>
  <xacro:include filename="$(find delta_robot_ros2)/urdf/base.xacro"/>
  <xacro:include filename="$(find delta_robot_ros2)/urdf/leg.xacro"/>
  <xacro:include filename="$(find delta_robot_ros2)/urdf/platform.xacro"/>
  <xacro:include filename="$(find delta_robot_ros2)/urdf/robot_control.xacro"/>

<!-- *********************************************************************** -->
  <!-- Параметры для основания и платформы -->
  <xacro:property name="base_radius" value="0.5"/>
  <xacro:property name="base_height" value="0.05"/>
  <xacro:property name="base_material_rgba" value="1 0 0 0.8"/>

  <xacro:property name="platform_radius" value="0.3"/>
  <xacro:property name="platform_height" value="0.05"/>
  <xacro:property name="platform_material_rgba" value="0 1 0 0.8"/>

  <xacro:property name="platform_initial_height" value="0.6"/>

<!-- *********************************************************************** -->

  <!-- Блок расчёта расположения и ориентации ног дельтаробота -->
  <xacro:property name="base_offset_z" value="0.2"/>
  <xacro:property name="angle_deg_1" value="0"/>
  <xacro:property name="angle_deg_2" value="120"/>
  <xacro:property name="angle_deg_3" value="240"/>

  <xacro:property name="leg_1_origin_xyz" value="${base_radius * cos(angle_deg_1 * pi / 180)} ${base_radius * sin(angle_deg_1 * pi / 180)} ${base_offset_z}"/>
  <xacro:property name="leg_1_origin_rpy" value="0 0 ${angle_deg_1 * pi / 180}"/>

  <xacro:property name="leg_2_origin_xyz" value="${base_radius * cos(angle_deg_2 * pi / 180)} ${base_radius * sin(angle_deg_2 * pi / 180)} ${base_offset_z}"/>
  <xacro:property name="leg_2_origin_rpy" value="0 0 ${angle_deg_2 * pi / 180}"/>

  <xacro:property name="leg_3_origin_xyz" value="${base_radius * cos(angle_deg_3 * pi / 180)} ${base_radius * sin(angle_deg_3 * pi / 180)} ${base_offset_z}"/>
  <xacro:property name="leg_3_origin_rpy" value="0 0 ${angle_deg_3 * pi / 180}"/>


<!-- *********************************************************************** -->

  <!-- Основание робота -->
  <xacro:base id="base" base_radius="${base_radius}" base_height="${base_height}" base_material_rgba="${base_material_rgba}"/>

  <!-- Манипулируемая платформа -->
  
  <xacro:platform id="platform" platform_radius="${platform_radius}" platform_height="${platform_height}" platform_material_rgba="${platform_material_rgba}" parent="base_link" origin_xyz="0 0 ${platform_initial_height}" origin_rpy="0 0 0"/>

  <!-- Ноги робота -->
  <xacro:leg prefix="leg_1" origin_xyz="${leg_1_origin_xyz}" origin_rpy="${leg_1_origin_rpy}"/>
  <xacro:leg prefix="leg_2" origin_xyz="${leg_2_origin_xyz}" origin_rpy="${leg_2_origin_rpy}"/>
  <xacro:leg prefix="leg_3" origin_xyz="${leg_3_origin_xyz}" origin_rpy="${leg_3_origin_rpy}"/>
<!-- *********************************************************************** -->

  <xacro:daleta_robot_control/>  

</robot>